src/transaction.rs

```diff
@@ -195,25 +269,33 @@ impl SigningProcess {
             }
 
             if self.own_public_key == pk
-                || self.other_commitments.iter().any(|c| c.public_key == pk)
+                || self.other_commitments_list.iter().any(|c| c.public_key == pk)
             {
                 println!("🤨 Duplicate public key, ignoring this one.");
                 continue;
             }
 
-            println!("  Enter the corresponding commitment:");
-            let commitment_str = read_line()?;
-            let mut commitment = [0u8; Commitment::SIZE];
-            let commitment = if let Ok(_) = hex::decode_to_slice(&commitment_str, &mut commitment) {
-                Commitment::from(commitment)
-            } else {
-                println!("🤨 Could not parse the commitment. Try again.");
-                continue;
-            };
+            println!("  Enter the corresponding commitment list:");
+            let mut i = 0;
+            let mut collected_commitment_list = vec![];
+            while i < MUSIG2_PARAMETER_V{
```

Comment:
We don't need i. You can remove it and replace it by the length of the list.
```suggestion
            while collected_commitment_list.len() < MUSIG2_PARAMETER_V {
```